internal/cli/opsmanager/admin/backup/s3/update.go

```diff
@@ -0,0 +1,139 @@
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/opsmanager/admin/backupstore"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+var updateTemplate = "S3 blockstore configuration '{{.ID}}' updated.\n"
+
+type UpdateOpts struct {
+	cli.OutputOpts
+	backupstore.AdminOpts
+	store                  store.S3BlockstoresUpdater
+	awsAccessKey           string
+	awsSecretKey           string
+	s3AuthMethod           string
+	s3BucketEndpoint       string
+	s3BucketName           string
+	s3MaxConnections       int64
+	disableProxyS3         bool
+	acceptedTos            bool
+	sseEnabled             bool
+	pathStyleAccessEnabled bool
+}
+
+func (opts *UpdateOpts) init() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *UpdateOpts) Run() error {
+	r, err := opts.store.UpdateS3Blockstores(opts.ID, opts.newS3Blockstore())
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) newS3Blockstore() *opsmngr.S3Blockstore {
+	blockstore := &opsmngr.S3Blockstore{
+		BackupStore:      *opts.NewBackupStore(),
+		AWSAccessKey:     opts.awsAccessKey,
+		AWSSecretKey:     opts.awsSecretKey,
+		S3AuthMethod:     opts.s3AuthMethod,
+		S3BucketEndpoint: opts.s3BucketEndpoint,
+		S3BucketName:     opts.s3BucketName,
+		S3MaxConnections: opts.s3MaxConnections,
+		AcceptedTos:      &opts.acceptedTos,
+	}
+
+	if opts.disableProxyS3 {
+		blockstore.DisableProxyS3 = &opts.disableProxyS3
+	}
+
+	if opts.sseEnabled {
+		blockstore.SSEEnabled = &opts.sseEnabled
+	}
+
+	if opts.pathStyleAccessEnabled {
+		blockstore.PathStyleAccessEnabled = &opts.pathStyleAccessEnabled
+	}
+
+	return blockstore
+}
+
+// mongocli ops-manager admin backup blockstore(s) update [--assignment][--encryptedCredentials][--id id]
+// [--label label][--loadFactor loadFactor][--uri uri][--ssl][--writeConcern writeConcern] [--awsAccessKey awsAccessKey]
+// [--awsSecretKey awsSecretKey] [--s3AuthMethod s3AuthMethod] [--s3BucketEndpoint s3BucketEndpoint] [--s3BucketName s3BucketName]
+// [--s3MaxConnections s3MaxConnections] [--disableProxyS3 disableProxyS3] [--acceptedTos acceptedTos] [--sseEnabled sseEnabled]
+// [--pathStyleAccessEnabled pathStyleAccessEnabled]
+
+func UpdateBuilder() *cobra.Command {
+	opts := &UpdateOpts{}
+	opts.Template = updateTemplate
+	cmd := &cobra.Command{
+		Use:   "update <blockstoreID>",
+		Args:  cobra.ExactArgs(1),
+		Short: update,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.OutWriter = cmd.OutOrStdout()
+			return opts.init()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.ID = args[0]
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.ID, flag.ID, "", usage.BlockstoreID)
```

Comment:
[nit] shouldn't the id be the `blockstoreID` of the command param?

Reply:
You moved this to a param instead of flag but kept the flag, and kept it as mandatory

Reply:
right. removed it